### DISCUSSION TITLE
[Serializer] Are the supported formats documented anywhere?

### DISCUSSION THREAD

**Discussion by PrOF-kk** [symfony/symfony#53008] [Q&A] [answered]
Aside from the "common sense" formats like `json`, `yaml`, `xml`, what are the possible values for the `$format` argument in
```php
$this->serializer->serialize($object, $format, $context);
$this->normalizer->normalize($object, $format, $context);
$this->encoder->encode($data, $format, $context);
```
?

For example, I found that `$this->normalizer->normalize($object, null, $context);`
and `$this->normalizer->normalize($object, 'associative', $context);` give different results, but I couldn't find any info on the `associative` format that I found by trial and error.

Thanks

**Comment by mbabker** [comment, marked as answer]
Out of the box there are CSV, JSON, XML, and YAML encoders, and you can add your own encoders.  Through the `SerializerInterface`, the format is needed so it knows what format to encode to when serializing and deserializing.  In practice, I'm not sure when folks would use the format argument on the `NormalizerInterface` and `DenormalizerInterface`.

Never seen that "associative" format to have any idea what to say there.

**Comment by jdevinemt** [comment]
The [documentation](https://symfony.com/doc/current/components/serializer.html#encoders) identifies the available formats in the ["Encoders"](https://symfony.com/doc/current/components/serializer.html#encoders) section. These formats are `json`, `xml`, `yaml` and `csv`. I don't see any reference to an `associative` format anywhere in the source code. I assume when you pass the `associative` value to format, it is just falling back to whatever the default behavior is when a matching encoder isn't found.

**Comment by PrOF-kk** [comment]
Looks like I con't replicate what I was seeing (null format gives different results compared to "any string" format) anymore.

Thanks for the help
